src/api/v1/folders.py

```diff
@@ -452,3 +448,223 @@ def delete_user_role(
+        session_id = session[1]
+
+        headers = {
+            "Content-Type": "application/json",
+            "Accept": "text/event-stream",
+        }
+
+        # Prepend instructions to the user prompt to limit access to the specified folder.
+        # The access to the folder is guarded by the backend system using access control, but we
+        # want to ensure the agent only accesses files in the specified folder.
+        # TODO: Build a more robust solution for this.
+        pre_prompt = f"ONLY access files in folder with ID: {folder_id}\n\n"
+
+        # Format the user prompt for the agent.
+        question_prompt = {
+            "role": "user",
+            "parts": [
+                {
+                    "text": pre_prompt + request.question_prompt,
+                }
+            ],
+        }
+
+        # Prepare the POST payload for the ADK server SSE stream.
+        payload = {
+            "appName": request.agent_name,
+            "userId": f"{current_user.id}",
+            "sessionId": session_id,
+            "newMessage": question_prompt,
+        }
+
+        # Start the SSE stream to the ADK server. The server will handle the agent execution
+        # and return the results as a stream of events.
+        try:
+            async with httpx.AsyncClient(timeout=None) as client:
+                async with client.stream("POST", url, headers=headers, json=payload) as response:
+                    response.raise_for_status()
+                    async for chunk in response.aiter_bytes():
+                        decoded_chunk = chunk.decode("utf-8")
+                        for line in decoded_chunk.splitlines():
+                            if line.startswith("data: "):
+                                try:
+                                    json_data = json.loads(line[len("data: ") :])
+                                    yield json.dumps(json_data)
+                                except json.JSONDecodeError:
+                                    pass
+                    yield json.dumps({"type": "complete", "message": "All agents finished."})
+        except httpx.HTTPStatusError as e:
+            error_message = (
+                f"HTTP error during SSE stream: {e.response.status_code} - {e.response.text}"
+            )
+            yield json.dumps({"type": "error", "message": error_message})
+        except httpx.RequestError as e:
+            error_message = f"Request error during SSE stream: {e}"
+            yield json.dumps({"type": "error", "message": error_message})
+        except Exception as e:
+            error_message = f"Unexpected error during SSE stream: {e}"
+            yield json.dumps({"type": "error", "message": error_message})
+
+    return EventSourceResponse(_run_sse_stream())
+
+
+@router.post("/{folder_id}/investigations/questions")
+@require_access(allowed_roles=[Role.VIEWER, Role.EDITOR, Role.OWNER])
+def get_investigative_questions(
```

Comment:
Probably move this to the agent itself in the future where the prompt would be a simple "Start a new investigation" and the agent itself has a template like the prompt here. This makes it easier to maintain and customize. The server can stay the same and we adapt and tune to agent itself to come up with the best starting state.

Reply:
Good point - yeah let's see where we want to do this in the future. Questions are going to be part of the database as well. This is easy to refactor in the future.